### DISCUSSION TITLE
SD build problems with Gordon's

### DISCUSSION THREAD

**Discussion by kbonin** [alanbjohnston/CubeSatSim#147] [Q&A] [answered]
Just wanted to pass this on... I tried to build a working SD image using the directions [here](https://github.com/alanbjohnston/CubeSatSim/wiki/Creating-the-CubeSatSim-Raspberry-Pi-Image) for a a Pi Zero 2 W, using the 64-bit Lite image from [here](https://downloads.raspberrypi.org/raspios_lite_arm64/images/raspios_lite_arm64-2022-04-07/2022-04-04-raspios-bullseye-arm64-lite.img.xz).

Ran into errors starting with this below and ending with no binaries generated
```bash
gcc -std=gnu99 -DDEBUG_LOGGING -o cubesatsim -Wall -Wextra -L./ afsk/ax25.o afsk/ax5043.o TelemEncoding.o main.o -lwiringPi -lax5043 -lm
/usr/bin/ld: skipping incompatible /usr/lib/gcc/aarch64-linux-gnu/10/../../../../lib/libwiringPi.so when searching for -lwiringPi
```
I'm not sure if the underlying problem I had is the drogon.net library is no longer maintained and/or doesn't work on the Pi 2 W or on 64-bit OS?  The one time I wandered down the wiring for pi rabbithole and ended up at http://wiringpi.com it seemed the original author had quit maintaining the library but didn't make it open source.  I was considering writing this thing myself (so I could use ardruino library conventions on the Pi with a little multithreading cleanup from modern c/c++) and was glad to find WiringPi but sad to see it wasn't open or maintained.  https://github.com/WiringPi/WiringPi seems to be the closest thing to a replacement?  It had a link to the now deleted deprecation announcement of WiringPi, but the relationship between the two is clear as mud.  In any case what fixed my build:
- sudo rm /lib/libwiringPi*
- install and build https://github.com/WiringPi/WiringPi
- comment out the lines in the CubeSatSim Makefile to wget and install the drogon copy

Wondering if I might have missed something?  (I did try multiple clean/rebuilds, but all the errors i ran into seemed to start with WiringPi problems.)

Thank you!

**Comment by alanbjohnston** [comment, marked as answer]
Hi! I wouldn’t recommend using the 64-bit version. It hadn’t been tested at all with the code, and neither have the libraries. 

I tried to get away completely from wiringpi since it was abandoned and instead use Python libraries, but it might still be used somewhere. I’ll try to remove it completely in a future release.
